### ISSUE TITLE
Assign experiment (fragmentation conditions) to spectra based on method XML

### ISSUE THREAD

**Issue by pavel-shliaha** [sgibb/topdownr#2]
When creating experiments I used modifications to assign a particular retention time window to a particular experiment, e.g. 

<Modification Order="291">
<Experiment ExperimentIndex="26">
<StartTime>13.01</StartTime>
<EndTime>13.5</EndTime>
</Experiment>
</Modification>

so the XML of course contains the time of each experiment and the settings used in this experiment.

When performing the acquisitions, time assigned for each experiment is calculated to be longer than time expected fragmentation to take for the experiment. E.g., if I calculate that time necessary to perform fragmentation for experiment 1 is 45sec I will make the experiment to be 1 min long. The rationale behind this is that the time it takes for a particular MS2 scan is unpredictable, since orbitrap  accumulates ions before injection. I.e. that a scan can take shorter or longer depending on when the  necessary amount of ions is accumulated.

Consider the situation when first MS2 experiment is assigned 1 min but the MS2 scan takes 45 sec. At 45sec the scan is over and at this point a second scan is initiated which will take further 45 sec, however MS2 experiment 2 starts at 60 sec. What happens is that second MS2 scan of experiment 1 will proceed until 90 sec and the first MS1 scan of MS2 experiment 2 will start at 90 sec. This is a problem since now the data contains two MS2 scans of experiment 1 and one MS2 scan of experiment 2. 

Basically what this means is that MS2 scan number cannot be correlated to a particular experiment. The only way to assign MS2 scan number to a particular experiment and hence fragmentation conditions is to look at the retention time, since the retention time at which scan starts actually deermines which experiment it corresponds to. E.g.

Experiment 1: 0.01-1min
Experiment 2: 1.01-2min

MS2 scan 1: 0-45sec - experiment 1
MS2 scan 2: 45.01-90sec - experiment 1
MS2 scan 3: 90.01sec - 135sec experiment 2

@sgibb please produce a function that takes the mzML of MS2 scans and XML of method and assigns mzML MS2 scan fragmentation conditions based on information of XML
